sound/soc/sof/pm.c

```diff
@@ -444,3 +444,27 @@ int snd_sof_suspend(struct device *dev)
 	return sof_suspend(dev, false);
 }
 EXPORT_SYMBOL(snd_sof_suspend);
+
+int snd_sof_prepare(struct device *dev)
+{
+	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
+
+#if defined(CONFIG_ACPI) && defined(CONFIG_SND_SOC_SOF_DEBUG_D0I3_AT_S0IX)
```

Comment:
The intention here is, 
1. some people might want the audio DSP to enter to D3(at S0ix) even the platform can support D0i3, so the the debug item is "D0i3 at S0ix".
2. eventually, I want to change the default of this item to be 'true', which means turn on "D0i3 at S0ix" if the platform supported. But to make it compatible at this moment for other platforms (to avoid possible massive regressions on some other platforms which are not tested yet).

So this is a flag for the transition period only, finally, it could be a true debug option "D3 at S0ix" which is selected for debug purpose only.

Reply:
@keyonjie I feel like the DEBUG_D0i3 is a bit of an overkill at this point. Lets leave this for later if really needed. For now, if the DSP supports D0i3, we should enter D0i3.

Reply:
OK, let me remove that, but we need to keep in mind that the suspend/resume behaviors will be changed for many other platforms, and new issues might be reported.